src/controller/datacontroller.js

```diff
@@ -216,19 +262,51 @@ export default class DataController {
 	 * This method also creates a batch with all the changes applied. If all you need is to parse data, use
 	 * the {@link #parse} method.
 	 *
-	 * @param {String} data Input data.
-	 * @param {String} [rootName='main'] Root name.
+	 * When data is passed as a string it is set on a default `main` root:
+	 *
+	 *		dataController.set( '<p>Foo</p>' ); // Sets data on the `main` root.
+	 *
+	 * To set data on a different root or multiple roots at once, object containing `rootName` - `data` pairs should be passed:
+	 *
+	 *		dataController.set( { main: '<p>Foo</p>', title: '<h1>Bar</h1>' } ); // Sets data on the `main` and `title` roots.
+	 *
+	 * @param {String|Object.<String,String>} data Input data as a string or an object containing `rootName` - `data`
+	 * pairs to set data on multiple roots at once.
 	 */
-	set( data, rootName = 'main' ) {
-		// Save to model.
-		const modelRoot = this.model.document.getRoot( rootName );
+	set( data ) {
+		let initialData = {};
```

Comment:
This is not `initialData` in this case. Maybe you can just use `data`?

Reply:
`data` is a param name already so I can go with something like:

```js
if ( typeof data === 'string' ) {
    data = { main: data }; // Default root is 'main'. To set data on a different root, object should be passed.
}
```

but it is a little bit ugly or try to come up with a new name.

Reply:
I changed it to `newData`.